pkg/catalogv2/git/git.go

```diff
@@ -1,480 +1,440 @@
+	r.cloneOpts.Tags = gogit.NoTags
+
+	// Fetch Options
+	r.fetchOpts.RemoteURL = r.URL
+	r.fetchOpts.Depth = 1
+	r.fetchOpts.Force = true
+	r.fetchOpts.InsecureSkipTLS = r.insecureTLSVerify
+	r.fetchOpts.Tags = gogit.NoTags
+
+	// Checkout Options
+	r.checkoutOpts.Force = true
+	r.checkoutOpts.Create = true
+
+	// List Options
+	r.listOpts.InsecureSkipTLS = r.insecureTLSVerify
+
+	// Reset Options
+	r.resetOpts.Mode = gogit.HardReset
+
+	// Set authentication Methods for cloning
+	if r.auth != nil {
+		r.cloneOpts.Auth = r.auth
+		r.fetchOpts.Auth = r.auth
+		r.listOpts.Auth = r.auth
+	}
+	// Set CABundle for all git operations
+	if len(r.caBundle) > 0 {
+		r.cloneOpts.CABundle = r.caBundle
+		r.fetchOpts.CABundle = r.caBundle
+		r.listOpts.CABundle = r.caBundle
+	}
+	// Debug if enabled
+	if logrus.IsLevelEnabled(logrus.DebugLevel) {
+		r.cloneOpts.Progress = os.Stdout
+		r.fetchOpts.Progress = os.Stdout
 	}
-
-	return g.fetchAndReset(branch)
 }
 
-func (g *git) httpClientWithCreds() (*http.Client, error) {
-	var (
-		username  string
-		password  string
-		tlsConfig tls.Config
-	)
-
-	if g.secret != nil {
-		switch g.secret.Type {
-		case corev1.SecretTypeBasicAuth:
-			username = string(g.secret.Data[corev1.BasicAuthUsernameKey])
-			password = string(g.secret.Data[corev1.BasicAuthPasswordKey])
-		case corev1.SecretTypeTLS:
-			cert, err := tls.X509KeyPair(g.secret.Data[corev1.TLSCertKey], g.secret.Data[corev1.TLSPrivateKeyKey])
-			if err != nil {
-				return nil, err
-			}
-			tlsConfig.Certificates = append(tlsConfig.Certificates, cert)
-		}
+// cloneOrOpen executes the clone operation of a git repository at given branch with depth = 1 if it does not exist.
+// If if exists, just open the local repository and assign it to ro.localRepo
+func (r *repository) cloneOrOpen(branch string) error {
+	var err error
+	cloneOptions := r.cloneOpts
+	if branch != "" {
+		cloneOptions.ReferenceName = plumbing.NewBranchReferenceName(branch)
 	}
 
-	if len(g.caBundle) > 0 {
-		cert, err := x509.ParseCertificate(g.caBundle)
-		if err != nil {
-			return nil, err
-		}
-		pool, err := x509.SystemCertPool()
-		if err != nil {
-			pool = x509.NewCertPool()
-		}
-		pool.AddCert(cert)
-		tlsConfig.RootCAs = pool
+	err = cloneOptions.Validate()
+	if err != nil {
+		return fmt.Errorf("plainClone validation failure: %w", err)
 	}
 
-	transport := http.DefaultTransport.(*http.Transport).Clone()
-	transport.TLSClientConfig = &tlsConfig
-	transport.TLSClientConfig.InsecureSkipVerify = g.insecureTLSVerify
+	err = r.plainOpen()
 
-	client := &http.Client{
-		Transport: transport,
-		Timeout:   30 * time.Second,
-	}
-	if username != "" || password != "" {
-		client.Transport = &basicRoundTripper{
-			username: username,
-			password: password,
-			next:     client.Transport,
+	if err == gogit.ErrRepositoryNotExists {
+		r.repoGogit, err = gogit.PlainClone(r.Directory, false, cloneOptions)
+		if err != nil && err != gogit.ErrRepositoryAlreadyExists {
```

Comment:
Sorry I did not understand what you meant.

Reply:
Sorry, I meant to comment on Rohits review. 
What I meant was that if the `r.plainOpen()` returns an `ErrRepositoryNotExists` but when you call the `gogit.PlainClone()` it returns an `ErrRepositoryAlreadyExists`, we should probably have a different error message since that would imply there's something wrong with `r.plainOpen()`